x-pack/plugins/actions/server/index.ts

```diff
@@ -57,7 +57,37 @@ export const plugin = (initContext: PluginInitializerContext) => new ActionsPlug
 
 export const config: PluginConfigDescriptor<ActionsConfig> = {
   schema: configSchema,
-  deprecations: ({ renameFromRoot }) => [
+  deprecations: ({ renameFromRoot, unused }) => [
     renameFromRoot('xpack.actions.whitelistedHosts', 'xpack.actions.allowedHosts'),
+    (settings, fromPath, addDeprecation) => {
```

Comment:
do we have any way of testing these?

Reply:
also, we already have some functional tests that test at least a bit of this stuff - can't test all the combinations, but I think what we have now is good enough; see https://github.com/elastic/kibana/pull/96630 for those tests.  We can either add more variants with the new options, or change some of the existing ones to use the new options, or both ...

Reply:
Remembering there are some slightly odd "jest tests" that got added in #96630 which actually start up TLS servers to make sure the configs will actually work with axios.  They are more like "function tests" in that they actually do make real requests, but they are heavily mocked, and allow us to run a lot more combinations of options that we can with the actual function tests.  In the file `x-pack/plugins/actions/server/builtin_action_types/lib/axios_utils_connection.test.ts`.  We should add some more tests there as well, with the new options ...

Reply:
@pmuellr I've changed integration tests to use the new verificationMode option and added a tests for legacy `rejectUnauthorized` global configuration. Do you think we need something in addition?

Reply:
I think this will be good enough, thx!